FlowCrypt/Controllers/Compose/ComposeViewController.swift

```diff
@@ -216,61 +218,65 @@ extension ComposeViewController {
+    private func encryptAndSendMessage() {
+        let recipients = contextToSend.recipients
 
-            let recipients = self.contextToSend.recipients
-
-            guard recipients.isNotEmpty else {
-                assertionFailure("Recipients should not be empty. Fail in checking")
-                return false
-            }
+        guard recipients.isNotEmpty else {
+            showAlert(message: "Recipients should not be empty. Fail in checking")
+            return
+        }
 
-            guard let text = self.contextToSend.message else {
-                assertionFailure("Text and Email should not be nil at this point. Fail in checking")
-                return false
-            }
+        guard let text = self.contextToSend.message else {
+            showAlert(message: "Text and Email should not be nil at this point. Fail in checking")
+            return
+        }
 
-            let subject = self.input.subjectReplyTitle
-                ?? self.contextToSend.subject
-                ?? "(no subject)"
+        let subject = self.input.subjectReplyTitle
+            ?? self.contextToSend.subject
+            ?? "(no subject)"
 
-            guard let myPubKey = self.dataService.publicKey() else {
-                self.showAlert(message: "compose_no_pub_sender".localized)
-                return false
-            }
+        guard let myPubKey = self.dataService.publicKey() else {
+            self.showAlert(message: "compose_no_pub_sender".localized)
+            return
+        }
 
-            guard let allRecipientPubs = self.getPubKeys(for: recipients) else {
-                return false
-            }
+        guard let allRecipientPubs = getPubKeys(for: recipients) else {
+            showAlert(message: "Public key is missin")
+            return
+        }
 
-            let encrypted = self.encryptMsg(
-                pubkeys: allRecipientPubs + [myPubKey],
-                subject: subject,
-                message: text,
-                to: recipients.map(\.email)
-            )
+        showSpinner()
 
-            try awaitPromise(self.messageSender.sendMail(mime: encrypted.mimeEncoded))
+        let encrypted = self.encryptMsg(
+            pubkeys: allRecipientPubs + [myPubKey],
+            subject: subject,
+            message: text,
+            to: recipients.map(\.email)
+        )
 
-            return true
-        }
+        messageSender
+            .sendMail(mime: encrypted.mimeEncoded)
+            .sink(
+                receiveCompletion: { [weak self] result in
+                    guard let error = result.getError() else {
+                        return
+                    }
+
+                    self?.showAlert(error: error, message: "compose_error".localized)
+                },
+                receiveValue: { [weak self] in
+                    self?.handleSuccessfullySentMessage()
+                })
+            .store(in: &cancellable)
```

Comment:
I'll file an issue to create a slightly different convention here for increased clarity.